src/app/_components/Form/FormFileInput.tsx

```diff
@@ -0,0 +1,179 @@
+import NextImage from 'next/image'
+
+import {
+  Field,
+  Input,
+  Description,
+  type InputProps as HeadlessInputProps,
+} from '@headlessui/react'
+import { Image, X } from '@phosphor-icons/react/dist/ssr'
+import { clsx } from 'clsx'
+
+import { FormError, type FormErrorProps } from '@/components/Form/FormError'
+import { FormLabel, type FormLabelProps } from '@/components/Form/FormLabel'
+import { Icon } from '@/components/Icon'
+
+type HardCodedProps = 'type' | 'onChange' | 'accept'
+
+export type FormFileInputProps = {
+  files: FileList | null
+  accept: Array<`.${string}`>
+  maxSize: number
+  description?: string | React.ReactNode
+  onChange: (files: FileList | null) => void
+} & Omit<HeadlessInputProps, HardCodedProps> &
+  FormLabelProps &
+  FormErrorProps
+
+export function FormFileInput({
+  files,
+  label,
+  hideLabel,
+  error,
+  hideError,
+  disabled,
+  onChange,
+  description,
+  maxSize,
+  accept,
+  ...rest
+}: FormFileInputProps) {
+  const loadedFile = files?.[0]
+
+  function resetFiles() {
+    onChange(null)
+  }
+
+  function loadFiles(e: React.ChangeEvent<HTMLInputElement>) {
+    const files = e.target.files
+
+    if (files) {
+      onChange(files)
+    }
```

Comment:
should we handle if the user uploads > MAX_SIZE?

```suggestion
 if (files && files.length > 0) {
    if (files[0].size > maxSize) {
      // Handle file size error
      console.error('File size exceeds maximum allowed size');
      return;
    }
    onChange(files)
  }
```